_adr/87/index.adoc

```diff
@@ -1,48 +1,230 @@
+**** Requester
+**** Control Plane team
+**** Kafka Integrations team
+
+. Expose the metric in the KAS Fleet Manager API
+** This should be assigned to the Control Plane team.
+** Any tasks for adding the new metric to the API should be done in this stage.
+This includes adding the new metric to the Prometheus remote-write configuration and any code changes to the KAS Fleet Manager repository.
+** [.underline]#Definition of Done#:
+*** New customer facing metric should be included in the response of the following KAS Fleet Manager API endpoints:
+**** https://api.openshift.com/?urls.primaryName=kafka%20service%20fleet%20manager%20service#/default/getMetricsByInstantQuery[/api/kafkas_mgmt/v1/kafkas/{id}/metrics/query]
+**** https://api.openshift.com/?urls.primaryName=kafka%20service%20fleet%20manager%20service#/default/getMetricsByRangeQuery[/api/kafkas_mgmt/v1/kafkas/{id}/metrics/query_range]
+**** https://api.openshift.com/?urls.primaryName=kafka%20service%20fleet%20manager%20service#/default/federateMetrics[/api/kafkas_mgmt/v1/kafkas/{id}/metrics/federate]
+
+. Document the new metric
+** This should be assigned to the Documentation/Customer Content Support team.
+** [.underline]#Definition of Done#:
+*** New metric documented in the app-services-guides: https://github.com/redhat-developer/app-services-guides/tree/main/docs/kafka/metrics-monitoring-kafka[Monitoring metrics in OpenShift Streams for Apache Kafka].
+
+
+### Migrating existing metrics to the new naming policy
+KAS Fleet Manager does not support API versioning at the moment so we would need to support both existing and new metrics during the migration process.
+Since the revised metric names will be added as new metrics, they will have no history.
+This will need to be taken into account during the migration.
+
+The Kafka console provides dashboards which can display data up to the last 7 days (see the design in https://redhat-developer.github.io/app-services-ui-components/?path=/story/kafka-metrics--all-ready[Storybook]).
+Once 7 days has passed, since the new metrics has been added, the existing and new metrics would have the same history for the UI.
+At this point, we can expose the new metrics in KAS Fleet Manager and mark the existing metrics as deprecated. 
+
+The existing metrics shall be marked as deprecated as per the MAS API deprecation policy.
+The deprecation will be communicated as follows:
+
+* The KAS Fleet Manager OpenAPI specification will document the deprecated metrics and their alternatives in each of the endpoint descriptions.
+* The metrics listed in the https://github.com/redhat-developer/app-services-guides/tree/main/docs/kafka/metrics-monitoring-kafka[Monitoring metrics in OpenShift Streams for Apache Kafka] documentation shall be marked as 'deprecated', pointing the users to use the newly revised metrics.
+* An email will be sent out to mas-devel and mas-all mailing list as per the deprecation policy.
+* Deprecation period shall be 60 days.
```

Comment:
Subject to change

Reply:
rhobs team got back to us confirming that there is nothing to worry about the current state of metrics ingestion in Observatorium. We are far off from hitting any limits. Therefore there should be no problem with keeping deprecated metrics for a minimum of 60 days. We could increase this number as well if needed.

Reply:
Is 60 days enough for deprecation period?

Reply:
So there are two areas to consider:
* a metric name is an API contract and therefore should not be deprecated unless it is really problematic
* a metric content is naturally time bound and we could consider that a deprecation time wound be > a the time out period of a metric entry

Reply:
Thanks @emmanuelbernard!

Reply:
It depends on what we really productize on the metrics side for users. 
Right now we provide and document to ur customers an Kafka Dashboard dashboard that does provide additional abstraction layer for integrators:

https://grafana.com/grafana/dashboards/15835

We say that when users use that dashboard they would get updates on the configuration automatically thus if something breaks on their side they can do update and review it

That is why I think we should move contract to the client side artifacts (dashboards, grafana example datasets and SDKs. 

Current dashboard and Kafka SDK does that well. 

The main remaining element for our team is missing is the separation of the http metrics. We need new tag in the openAPI file to provide dedicated prometheus http SDK for our end users.

Reply:
> Discuss and figure out how prometheus "api" would be versioned.

This is listed as a non-goal. See https://github.com/bf2fc6cc711aee1a0c2a/architecture/pull/64#discussion_r887009810

> Figure out where the list of metrics is and likely point to it from the OpenAPI is

Out of scope of this ADR. JIRA created for this task: [MGDSTRM-9106](https://issues.redhat.com/browse/MGDSTRM-9106)